packages/@aws-cdk/aws-pipes-sources-alpha/lib/enums.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Define how to handle item process failures.
+ */
+export enum OnPartialBatchItemFailure {
```

Comment:
Do we expect this to have more values in the future? I don't see the benefit of using an enum here as is.

Reply:
I'm not sure on this. Possible AWS decides to add another method in the future?

Reply:
We did it this way for future-proofing. We don't have plans to expand the functionality beyond "automatic_bisect" but wanted to leave the option open to permit it. Once an API is delivered, it's extremely hard to change it. In the CDK world, if changing things in the future is also considered difficult, go with the enum... if it's easy to change it to an enum later, you are probably good with something simpler.

Reply:
Ok I'd prefer to leave this as an enum too.